File: main/src/main/java/net/fwitz/math/main/binary/complex/functions/exponential/LogZ.java
package net.fwitz.math.main.binary.complex.functions.exponential;

import net.fwitz.math.binary.complex.Complex;
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot;

public class LogZ {
    public static void main(String[] args) {
        new ComplexFunctionPlot("log(z)")
                .computeFn(Complex::log)
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/complex/functions/exponential/LogZ.kt
package net.fwitz.math.main.binary.complex.functions.exponential

import net.fwitz.math.binary.complex.Complex
import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot
import net.fwitz.math.plot.binary.complex.color.DomainColoringAdvancedPlusLogScale

object LogZ {
    @JvmStatic
    fun main(args: Array<String>) = ComplexFunctionPlot("log(z)")
        .computeFn(Complex::log)
        .colorFn(DomainColoringAdvancedPlusLogScale)
        .render()
}